opensrp-reveal/src/main/java/org/smartregister/reveal/interactor/BaseFormFragmentInteractor.java

```diff
@@ -81,5 +91,34 @@ public void findMemberDetails(String structureId, JSONObject formJSON) {
         });
     }
 
+    @Override
+    public void findSprayDetails(String interventionType, String structureId, JSONObject formJSON) {
+        if (IRS.equals(interventionType)) {
+
+            appExecutors.diskIO().execute(() -> {
+                Cursor cursor = null;
+                List<CommonPersonObject> list = new ArrayList<>();
+                try {
+                    cursor = eventClientRepository.getWritableDatabase().rawQuery(
+                            String.format("select s.*, id as _id from %s s where %s = ?", SPRAYED_STRUCTURES, BASE_ENTITY_ID), new String[]{structureId});
+                    if (cursor.moveToFirst()) {
+                        CommonPersonObject commonPersonObject = commonRepository.getCommonPersonObjectFromCursor(cursor);
+                        list.add(commonPersonObject);
```

Comment:
There is no need to add a list since `cursor.moveToFirst()` will only pick the 1st record returned